src/iterators/inverted_index_iterator.c

```diff
@@ -556,13 +556,14 @@ QueryIterator *NewInvIndIterator_TermQuery(const InvertedIndex *idx, const Redis
   record->fieldMask = RS_FIELDMASK_ALL;
   record->freq = 1;
 
-  IndexDecoderCtx dctx = {0};
-  if (fieldMaskOrIndex.isFieldMask && (InvertedIndex_Flags(idx) & Index_WideSchema))
-    dctx.wideMask = fieldMaskOrIndex.value.mask;
-  else if (fieldMaskOrIndex.isFieldMask)
-    dctx.mask = fieldMaskOrIndex.value.mask;
-  else
-    dctx.wideMask = RS_FIELDMASK_ALL; // Also covers the case of a non-wide schema
+  IndexDecoderCtx dctx = {.tag = IndexDecoderCtx_FieldMask};
+  if (fieldMaskOrIndex.isFieldMask && (InvertedIndex_Flags(idx) & Index_WideSchema)) {
+    dctx.field_mask = fieldMaskOrIndex.value.mask;
+  } else if (fieldMaskOrIndex.isFieldMask) {
+    dctx.field_mask = fieldMaskOrIndex.value.mask;
```

Comment:
So we longer have to differenciate between `mask` and `wideMask`?

If that's so then those `if` can be simplified.

Reply:
Nice catch!! That's one less FFI call needed. Done in https://github.com/RediSearch/RediSearch/pull/6760/commits/ff763af24b623db5e5a56d92c786ac95c5d47119